extensions/amp-audio/1.0/component.js

```diff
@@ -0,0 +1,151 @@
+ * limitations under the License.
+ */
+
+import * as Preact from '#preact';
+import {ContainWrapper} from '#preact/component';
+import {forwardRef} from '#preact/compat';
+import {
+  EMPTY_METADATA,
+  setMediaSession,
+} from '../../../src/mediasession-helper';
+import {
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+} from '#preact';
+
+/**
+ * @param {!AudioDef.Props} props
+ * @param {{current: (!AudioDef.AudioApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+export function AudioWithRef(
+  {
+    album = EMPTY_METADATA.album,
+    'aria-describedby': ariaDescribedby,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby,
+    artist = EMPTY_METADATA.artist,
+    artwork = EMPTY_METADATA.artwork,
+    autoplay = false,
+    children,
+    controlsList,
+    loop = false,
+    muted = false,
+    onPause,
+    onPlay,
+    preload,
+    sources,
+    src,
+    title = EMPTY_METADATA.title,
+    validateMediaMetadata,
+    ...rest
+  },
+  ref
+) {
+  const audioRef = useRef(null);
+  const wrapperRef = useRef(null);
+
+  /** @public {boolean} */
+  const isPlaying = useRef(false);
+
+  /**
+   * Prepares Media Metadata
+   */
+  const metaData = useMemo(() => {
+    return {
+      title,
+      artist,
+      album,
+      artwork: [{src: artwork}],
+    };
+  }, [title, artist, album, artwork]);
+
+  /**
+   * Plays audio callback
+   */
+  const play = useCallback(() => {
+    onPlay?.();
+    audioRef.current.play();
+    isPlaying.current = true;
+  }, [onPlay]);
+
+  /**
+   * Pauses audio callback
+   */
+  const pause = useCallback(() => {
+    onPause?.();
+    audioRef.current.pause();
+    isPlaying.current = false;
+  }, [onPause]);
+
+  /**
+   * Updates media session for current window/tab
+   */
+  const audioPlaying = useCallback(() => {
+    const win = audioRef.current?.ownerDocument?.defaultView;
+    const element = audioRef.current;
+
+    validateMediaMetadata?.(element, metaData);
+
+    setMediaSession(win, metaData, play, pause);
+  }, [metaData, validateMediaMetadata, play, pause]);
+
+  useEffect(() => {
+    if (!audioRef.current.play) {
+      return;
+    }
+  }, []);
+
+  /** Audio Component - API Functions */
+  useImperativeHandle(
+    ref,
+    () =>
+      /** @type {!AudioDef.AudioApi} */ ({
+        play,
+        pause,
+        isPlaying: () => isPlaying.current,
+      }),
+    [play, pause]
+  );
+
+  return (
+    <ContainWrapper contentRef={wrapperRef} size layout paint {...rest}>
+      <audio
+        ref={audioRef}
+        aria-describedby={ariaDescribedby}
+        aria-label={ariaLabel}
+        aria-labelledby={ariaLabelledby}
+        autoplay={autoplay}
+        controls // Force controls otherwise there is no player UI.
+        controlsList={controlsList}
+        loop={loop}
+        muted={muted}
+        onPlaying={() => audioPlaying()}
+        preload={preload}
+        src={src}
```

Comment:
This requires `loading="lazy"`. To follow convention, let's add a `loading` prop that defaults to `'lazy'`, and add:

```suggestion
        loading={loading}
```